Wikidata

almudenasanz opened the pull request:
### Description
The final objetive if to use Wikidata as a new Knowledge Graph for Recommendation algorithms, and to extract entities description to use new datasets (like Movielens) with DKN in DKN. This is the first step in that direction. I have implemented:

New utils functions to do specific queries in Wikidata:
- Query list of related entities from a string representing the name of an entity. The goal is to be able to create a Knowledge Graph from the linked entities in Wikidata
- Query entity description string representing the name of an entity

To test the functions created I have added a new notebook. The first section consists on creating a Knowledge Graph from the linked entities in Wikidata, and visualising the result of the KG. The second  part tests the enriching of the name of an entity with their description and list of related entities, the goal is using this enriching for new datasets (like Movielens) with DKN. 

### Related Issues
#525 

### Checklist:
- [x] My code follows the code style of this project, as detailed in our [contribution guidelines](../CONTRIBUTING.md).
- [x] I have added tests. -> I have added tests in the notebook, should I add more?
- [x] I have updated the documentation accordingly.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/recommenders?pullRequest=778) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/recommenders?pullRequest=778) before we can accept your contribution.<br/><br/>:x: almudenasanz [sign now](https://cla.opensource.microsoft.com/microsoft/recommenders?pullRequest=778)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/recommenders?pullRequest=778) it.</sub>

yueguoguo commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,102 @@
+        entityID: wikidata entityID corresponding to the title string. 
+                  'entityNotFound' will be returned if no page is found
+    """
+    url = "https://en.wikipedia.org/w/api.php?action=query&prop=pageprops&format=json&titles="
```
Hardcoded string may not be desirable. Make it an input variable or constant.

almudenasanz replied:
hi, sorry for the delay in the answer. since the request to the new query (I had to do some changes) is a concatenation of strings and a variable:

`requests.get("https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch="+name+"&format=json&prop=pageprops&ppprop=wikibase_item")` 

how do you suggest I can do this? should make the two substrings constants and concatenate them in the function?

gramhagen replied:
here's an approach
```
# defined at beginning of module
import urllib
API_URL = "https://en.wikipedia.org/w/api.php"


def find_wikidataID(name):
    url_opts = "&".join([
        "action=query",
        "list=search",
        "srsearch={}".format(urllib.parse.quote(name)),
        "format=json",
        "prop=pageprops",
        "ppprop=wikibase_item",
    ])
    requests.get("{url}?{opts}".format(url=API_URL, opts=url_opts))
```

almudenasanz replied:
I implemented this here: a5338c0 Thanks a lot for the tip!

yueguoguo commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,102 @@
+    """
+    query = """
+    PREFIX entity: <http://www.wikidata.org/entity/>
+    #partial results
```
Would it be better to put this into `Example` session of the docstring of the function?

almudenasanz replied:
The function itself is to perform this query with these parameters. If I move the query as an Example, I would need another function to indicate the value of the query
Maybe I'm not understanding it correctly, I would really appreciate it of you could clarify :)

gramhagen replied:
i think maybe there was just some confusion as they query is rather complicated it looks like a docstring? I don't think changes are needed here.

yueguoguo commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,102 @@
+        description (str): Wikidata short description of the entityID
+                           'descriptionNotFound' will be returned if no 
+                           description is found
+    """
```
Same as above - it might be better to make them `Example` in the docstring.

yueguoguo commented on notebooks/01_prepare_data/wikidata.ipynb:
```diff
@@ -0,0 +1,369 @@
+{
```
Add a good title of the notebook 😄

yueguoguo commented on notebooks/01_prepare_data/wikidata.ipynb:
```diff
@@ -0,0 +1,369 @@
+    "import sys\n",
+    "sys.path.append(\"../../\")\n",
+    "\n",
+    "import pandas as pd\n",
```
Add 
```
print("System version: {}".format(sys.version))
print("Spark version: {}".format(pyspark.__version__))
```
for system environment checking. `Spark` checking is not needed in this notebook.

yueguoguo commented on notebooks/01_prepare_data/wikidata.ipynb:
```diff
@@ -0,0 +1,369 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Create a KG from linked entities in Wikidata from a list"
```
I expect some text explanations at the beginning of the notebook about knowledge graph, basic concepts like entities, Wikidata, etc.

almudenasanz replied:
Added the info in the first cell :)

yueguoguo commented on notebooks/01_prepare_data/wikidata.ipynb:
```diff
@@ -0,0 +1,369 @@
+   "outputs": [],
+   "source": [
+    "import networkx as nx\n",
+    "import matplotlib.pyplot as plt\n",
```
The graph is so nice. Do we want to show the names instead of the entity ids?

almudenasanz replied:
changed this to show the names

miguelgfierro commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,102 @@
+
+def find_wikidataID(name):
+    """Find the entity ID in wikidata from a title string.
+    Args:
```
can you put a new line between Args and Returns? if not sphinx won't parse this. It should be:
```
"""Find the entity ID in wikidata from a title string.
    
Args:
    ....

Returns:
   ....
  
 
```

almudenasanz replied:
done

miguelgfierro commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,102 @@
+    Args:
+        entityID (str): A wikidata page ID.
+    Returns:
+        description (str): Wikidata short description of the entityID
```
there is no need to add the name of the variable, this would be:

```
Returns:
    str: Wikidata...

```

almudenasanz replied:
done

miguelgfierro commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,102 @@
+    LIMIT 500
+    """
+    url = 'https://query.wikidata.org/sparql'
+    r = requests.get(url, params = {'format': 'json', 'query': query})
```
is there a query limit for this? if so, we might want to use lru_cache

almudenasanz replied:
In the Query Limits section (https://www.mediawiki.org/wiki/Wikidata_Query_Service/User_Manual) of the documentation they mention: "the service is limited to 5 parallel queries per IP". Seems like the only limit is in parallel queries per IP, not in sequential queries

miguelgfierro commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,102 @@
+    try:
+        entityID = r.json()["query"]["pages"][entityID]["pageprops"]["wikibase_item"]
+    except:
+        entityID = "entityNotFound"
```
why returning a string vs raising an exception?

almudenasanz replied:
I return a string for the cases where the function is used in a loop, so we can have a record of that entity not having a response and the code can keep running. Do you recommend doing something else?

gramhagen replied:
this can vary a bit, but often I find it is easier to raise the exception here and let the calling function catch it and handle it as needed. This removes the need to check the output for a specific string defined here and so there is looser coupling across functions and more flexibility in handling errors upstream.

miguelgfierro commented:
> In the repository, we have developed a DevOps pipeline that helps test codes in the utility functions and notebooks. Try to understand how it works and how to write good unit tests for your functions. Goo examples can be found in the folder of tests.

@almudenasanz let me know if you need help with this. There is information here https://github.com/microsoft/recommenders/tree/master/tests

miguelgfierro commented:
Hi @almudenasanz, we were talking internally. @chenhuims is going to work KG networks with you. He will probably do CKE, which will complement your work on Ripple and KGCN. 

We thought that it would be interesting that the final output of the notebook that you are doing could be the knowledge graph of Movielens as a dumped file (in the format of the networks), which we would save in a blob. Then in the notebooks of the KG networks, we will start from that saved file. After that, you guys could work in parallel.

We would need to have a KG for the 4 ML (100k, 1M, 10M, 20M). @Leavingseason was wondering if there is a restriction in the number of requests in the Wikidata API

miguelgfierro commented:
One note on ways of working for @almudenasanz  and @chenhuims. In other situations where there are several people working on the same issue, they will agree on the tasks to work on (depending on the bandwidth) and then either push to the same branch (in your case it would be `wikidata` or one person will do a PR to that branch).

almudenasanz commented:
> 
> We would need to have a KG for the 4 ML (100k, 1M, 10M, 20M). @Leavingseason was wondering if there is a restriction in the number of requests in the Wikidata API

@Leavingseason @miguelgfierro In the [**Query Limits**] section (https://www.mediawiki.org/wiki/Wikidata_Query_Service/User_Manual) of the documentation they mention: "the service is limited to 5 parallel queries per IP". Seems like the only limit is in parallel queries per IP, not in sequential queries

gramhagen commented:
just curious if this would be made simpler by leveraging an existing package like: https://pypi.org/project/Wikidata/

miguelgfierro commented:
> just curious if this would be made simpler by leveraging an existing package like: https://pypi.org/project/Wikidata/

The latest version if from 2017, is this maintained?

gramhagen commented:
hmm, there are comments from the maintainer on issues that are more recent Dec 2018, it's possible the api hasn't changed enough to warrant any updates?

almudenasanz commented:
> just curious if this would be made simpler by leveraging an existing package like: https://pypi.org/project/Wikidata/

Hi @gramhagen , I looked into the package but for the specific implementations that I use on the notebook:
- Retrieve wikipedia page title from search query
- Retrieve wikidata item identifier from page title
- Retrieve all connected Wikidata item identifier from a Wikidata item identifier

I did not find implementations in the package that are simpler than the one's I implemented. But happy to discuss any suggestions!

almudenasanz commented:
> We would need to have a KG for the 4 ML (100k, 1M, 10M, 20M). @Leavingseason was wondering if there is a restriction in the number of requests in the Wikidata API

@miguelgfierro I added to the notebook a new section that extracts the entities for the 100k Movielens version. 

I had to reimplement a bit the _find_wikidataID_ method, because I was matching strings to exact wikipedia page titles. Some of the movie titles did not match exactly the wikipedia title, so I added a simple query that uses a text query to retrieve the first matching page title, and now works well for all movies.

I need to ask you where to put the output file of the MovieLens KG

gramhagen commented:
ok, no problem, wasn't sure if it would be helpful to leverage that, but sounds like it's not in this case. thanks for checking into it

miguelgfierro commented:
> I need to ask you where to put the output file of the MovieLens KG

we are changing the blobs of recommenders, can you send me the file somehow? then I'll uploaded to the correct place

miguelgfierro commented:
hey @almudenasanz @chenhuims would you please have an update on the state of this PR and the work you are doing with movielens+wikidata?

Please let me know if you have any blockers

almudenasanz commented:
Hi @miguelgfierro 

Chenhui and me talked one week ago and discussed the status of the project and how we could collaborate. We decided the next steps would be:

- @chenhuims will work on using the Wikidata Movieles KG with KGCN
- I will work on using the Wikidata MovieLens KG will Ripple
- We still need to understand how to prepare the Movielens dataset to use DKN. From my understanding, DKN works with [candidate news](https://github.com/Microsoft/Recommenders/blob/master/notebooks/00_quick_start/dkn_synthetic.ipynb) and then ranks which one is more likely to be selected. However, in Movielens we have movie ratings with a numerical score per movie per user, so we either transform this into candidate movies or we understand how to use DKN with Movielens. If you could help us figure this out, it would be great

miguelgfierro commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,105 @@
+        entityID: wikidata entityID corresponding to the title string. 
+                  'entityNotFound' will be returned if no page is found
+    """
+    r = requests.get("https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch="+name+"&format=json&prop=pageprops&ppprop=wikibase_item")
```
do you mind to explain what are these requests doing, and why are you chaining them?

almudenasanz replied:
Sure! In the previous version I was matching a string (the name of the movie) with the title of a wikipedia page, this was an issue for ambiguous names (eg. Brazil movie vs. Brazil the country). I added a previous request that receives a free text string and returns the first page found in a wikipedia, for Movielens I send "MovieTitle, MovieYear, film" and I then use the response with the wikipedia page title to query Wikidata

miguelgfierro commented:
> If you could help us figure this out, it would be great

There is some information in the KGCN paper: "Since the three datasets are explicit feedbacks, we transform
them into implicit feedback where each entry is marked with 1 indicating that the user has rated the item positively, and sample an unwatched set marked as 0 for each user. The threshold of positive rating is 4 for MovieLens-20M"

Something that is not clear to me, who will work on DKN? would it be you Almudena?

The output of this notebook was supposed to be used by the KG networks. Is this file `"movielens_100k_wikidata.csv"` what you guys need?

almudenasanz commented:
> There is some information in the KGCN paper: "Since the three datasets are explicit feedbacks, we transform
> them into implicit feedback where each entry is marked with 1 indicating that the user has rated the item positively, and sample an unwatched set marked as 0 for each user. The threshold of positive rating is 4 for MovieLens-20M"
Great, I will look into the code in KGCN

> Something that is not clear to me, who will work on DKN? would it be you Almudena?
Yes, I will work on it, I just need some time to figure out the data-wrangling part

> The output of this notebook was supposed to be used by the KG networks. Is this file `"movielens_100k_wikidata.csv"` what you guys need?
Yes, that is the output. With the notebook we can also recreate the output for other movielens datasets, we just need to change the movielens dataset loaded

miguelgfierro commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,105 @@
+    for c in data["results"]["bindings"]:
+        related_entities.append(c["valUrl"]["value"].replace("http://www.wikidata.org/entity/", ""))
+        related_names.append(c["valLabel"]["value"])
+    return (related_entities, related_names)
```
there is no need to use parenthesis here, this would be `return related_entities, related_names`

miguelgfierro commented on notebooks/01_prepare_data/wikidata.ipynb:
```diff
@@ -0,0 +1,567 @@
+{
```
<p>in the other notebooks we put the imports on the first cell</p>

 _Reply via <a href='https://app.reviewnb.com/microsoft/recommenders/pull/778/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

almudenasanz replied:
Implemented here: 9c03398

miguelgfierro commented on notebooks/01_prepare_data/wikidata.ipynb:
```diff
@@ -0,0 +1,567 @@
+{
```
<p>in the other notebooks we have an initial cell where we put the `MOVIELENS_DATA_SIZE`, see here: <a href="https://github.com/microsoft/recommenders/blob/master/notebooks/00_quick_start/sar_movielens.ipynb" target="_blank">https://github.com/microsoft/recommenders/blob/master/notebooks/00_quick_start/sar_movielens.ipynb</a>. We use that cell to programatically inject variables through papermill  for the tests. See  <a href="https://github.com/microsoft/recommenders/tree/master/tests#how-to-create-tests-on-notebooks-with-papermill" target="_blank">https://github.com/microsoft/recommenders/tree/master/tests#how-to-create-tests-on-notebooks-with-papermill</a></p>

 _Reply via <a href='https://app.reviewnb.com/microsoft/recommenders/pull/778/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

almudenasanz replied:
Implemented here: 9c03398

miguelgfierro commented on notebooks/01_prepare_data/wikidata.ipynb:
```diff
@@ -0,0 +1,567 @@
+{
```
<p>here we could have a filename that depends on the dataset size. This could be defined in the parameters cell</p>

 _Reply via <a href='https://app.reviewnb.com/microsoft/recommenders/pull/778/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

almudenasanz replied:
Implemented here: 9c03398

miguelgfierro commented on notebooks/01_prepare_data/wikidata.ipynb:
```diff
@@ -0,0 +1,567 @@
+{
```
<p>adding rows with concat is very inneficient, it is faster to use loc. See an example here: <a href="https://github.com/miguelgfierro/pybase/blob/1ce06734e236fc5367c6b946b16e4820b0cd84d0/pandas_base/conversion.py#L192" target="_blank">https://github.com/miguelgfierro/pybase/blob/1ce06734e236fc5367c6b946b16e4820b0cd84d0/pandas_base/conversion.py#L192</a></p>

 _Reply via <a href='https://app.reviewnb.com/microsoft/recommenders/pull/778/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

miguelgfierro commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,105 @@
+            entity_id = r.json()["query"]["pages"][str(pageID)]["pageprops"]["wikibase_item"]
+        except:
+            entity_id = "entityNotFound"
+    except:
```
I see that you are returning entity not found in some cases. You might want to name the except instead of using the generic value. See more info here: https://stackoverflow.com/a/16511493/5620182.

almudenasanz replied:
Int his commit I changed the exceptions to catch requests exceptions and used .get for the dictionaries: 51b0fd2

almudenasanz commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,105 @@
+        pageID = r.json()["query"]["search"][0]["pageid"]
+        r = requests.get("https://en.wikipedia.org/w/api.php?action=query&prop=pageprops&format=json&pageids="+str(pageID))
+        try:
+            entity_id = r.json()["query"]["pages"][str(pageID)]["pageprops"]["wikibase_item"]
```
perfect, I will change it

almudenasanz replied:
In this commit I changed the exceptions to catch requests exceptions and used .get for the dictionaries: 51b0fd2

gramhagen commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,102 @@
+        description = r.json()["results"]["bindings"][0]["o"]["value"]
+    except:
+        description = "descriptionNotFound"
+    return description
```
you can run [black](https://github.com/python/black) on this after you've finished changes, this will format the code to match the style used in the repo.

gramhagen commented on reco_utils/dataset/wikidata.py:
```diff
@@ -0,0 +1,102 @@
+    """
+    url = "https://en.wikipedia.org/w/api.php?action=query&prop=pageprops&format=json&titles="
+    r = requests.get(url+name.replace(" ", "%20"))
+    entityID = list(r.json()["query"]["pages"].keys())[0]
```
it's possible here too that the result may not be formed as expected, particularly if it is an error, this will also generate a TypeError or KeyError exception which calling functions will need to handle.

almudenasanz replied:
In this commit I changed the exceptions to catch requests exceptions and used .get for the dictionaries: 51b0fd2

miguelgfierro commented:
hey @almudenasanz, do you have any update with this?

almudenasanz commented:
hi @miguelgfierro I have been working on the data wrangling to create the type of files that Ripple uses: https://github.com/hwwang55/RippleNet/tree/master/data/movie from the outputs I generate in the notebook 
Then I will try to run the model with those first files locally

miguelgfierro commented:
> hi @miguelgfierro I have been working on the data wrangling to create the type of files that Ripple uses: https://github.com/hwwang55/RippleNet/tree/master/data/movie from the outputs I generate in the notebook
Then I will try to run the model with those first files locally

Would you be able to finish this PR first? there are some comments from members of the team. When you are finish I can compute the knowledge graphs of all the movielens and host them in a blob. This can be then used as input in the other models

miguelgfierro commented:
this looks good @almudenasanz. One question, how long does it take to compute the small KG with movielens 100k?

Can you add a test for the notebook? Depending on the time, they will go in unit test or maybe smoke tests

Here info on how to add the test https://github.com/microsoft/recommenders/tree/master/tests#how-to-create-tests-on-notebooks-with-papermill

miguelgfierro commented:
> so I was able to query the 100K Movielens dataset containing 1682 movies in 45 mins.

wow that's a lot. Maybe we can think of a way of running the first part (without movielens) in the unit tests, and then do some queries (not all) in the movielens part. Ideally we want the unit test to be less than one min and integration less than 15-20min.

Do you have any idea on how to execute the notebook under the times I mentioned?

almudenasanz commented:
> > so I was able to query the 100K Movielens dataset containing 1682 movies in 45 mins.
> 
> wow that's a lot. Maybe we can think of a way of running the first part (without movielens) in the unit tests, and then do some queries (not all) in the movielens part. Ideally we want the unit test to be less than one min and integration less than 15-20min.
> 
> Do you have any idea on how to execute the notebook under the times I mentioned?

I can create a parameter to run the tests only on a sample of the movielens dataset, would that work?

miguelgfierro commented:
> I can create a parameter to run the tests only on a sample of the movielens dataset, would that work?

yeah that's reasonable. I think a good way of doing this would be to have a unit test that just check that the first part of the notebook runs, similarly to [this example](https://github.com/microsoft/recommenders/blob/master/tests/unit/test_notebooks_python.py#L27). Then we can have what you suggested in the integration test and check programmatically some outputs [like in this example](https://github.com/microsoft/recommenders/blob/master/tests/integration/test_notebooks_python.py#L81). Maybe check that the first rows of the KG of movielens are correctly created. Kind of similar to what we are doing in [the criteo tests](https://github.com/microsoft/recommenders/blob/master/tests/integration/test_criteo.py#L15). 

Under this schema, maybe the integration tests take less than 5min

almudenasanz commented:
> yeah that's reasonable. I think a good way of doing this would be to have a unit test that just check that the first part of the notebook runs, similarly to [this example](https://github.com/microsoft/recommenders/blob/master/tests/unit/test_notebooks_python.py#L27). Then we can have what you suggested in the integration test and check programmatically some outputs [like in this example](https://github.com/microsoft/recommenders/blob/master/tests/integration/test_notebooks_python.py#L81).

Great, I already started working on them, and querying the 1M MovieLens dataset. I will commit them when I finish

almudenasanz commented:
I added the tests, and sent you the 1M file by email

I introduced sample parameters so both the unit and integration tests can run in under 1 min, and the integration test checks that the output os the file has the expected number of responses

miguelgfierro commented:
@almudenasanz, I solved the conflicts but one test failed, there is a small error in the code:
```
E           /anaconda/envs/reco_base/lib/python3.6/site-packages/tqdm/_tqdm.py in wrapper(*args, **kwargs)
E               673                     # take a fast or slow code path; so stop when t.total==t.n
E               674                     t.update(n=1 if not t.total or t.n < t.total else 0)
E           --> 675                     return func(*args, **kwargs)
E               676 
E               677                 # Apply the provided function (in **kwargs)
E           
E           <ipython-input-19-6ccb9974139b> in <lambda>(x)
E                 1 tqdm().pandas(desc="Number of movies completed")
E           ----> 2 result = pd.concat(list(movies.progress_apply(lambda x: wikidata_KG_from_movielens(x), axis=1)))
E           
E           <ipython-input-17-3cd273a9c115> in wikidata_KG_from_movielens(df)
E                 3     entity_id = find_wikidataID(df["Title"] + " film")
E                 4     if entity_id != "entityNotFound":
E           ----> 5         json_links = query_entity_links(entity_id)
E                 6         related_entities,related_names = read_linked_entities(json_links)
E                 7         d = pd.DataFrame({"original_entity":[entity_id]* len(related_entities),
E           
E           /data/home/recocat/cicd/3/s/reco_utils/dataset/wikidata.py in query_entity_links(entityID)
E               100         data = r.json()
E               101     except:
E           --> 102         print(e)
E               103         print("Entity ID not Found in Wikidata")
E               104         return {}
E           
E           NameError: ("name 'e' is not defined", 'occurred at index 21')

```

miguelgfierro commented:
@almudenasanz there is a time out:

```
$ pytest --durations=0 tests/unit/ -m "notebooks and not spark and not gpu"
======================================================= test session starts ========================================================
platform linux -- Python 3.6.8, pytest-4.2.1, py-1.7.0, pluggy-0.8.1
rootdir: /data/home/recocat/notebooks/miguel/Recommenders, inifile:
collected 163 items / 153 deselected / 10 selected

tests/unit/test_notebook_utils.py .                                                                                          [ 10%]
tests/unit/test_notebooks_python.py .........                                                                                [100%]

========================================================= warnings summary =========================================================
/anaconda/envs/reco_base/lib/python3.6/site-packages/tensorflow/python/util/tf_inspect.py:75
/anaconda/envs/reco_base/lib/python3.6/site-packages/tensorflow/python/util/tf_inspect.py:75
  /anaconda/envs/reco_base/lib/python3.6/site-packages/tensorflow/python/util/tf_inspect.py:75: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
    return _inspect.getargspec(target)

reco_utils/recommender/ncf/dataset.py:321
  /data/home/recocat/notebooks/miguel/Recommenders/reco_utils/recommender/ncf/dataset.py:321: DeprecationWarning: invalid escape sequence \[
    """

/anaconda/envs/reco_base/lib/python3.6/site-packages/nbconvert/exporters/exporter_locator.py:28
  /anaconda/envs/reco_base/lib/python3.6/site-packages/nbconvert/exporters/exporter_locator.py:28: DeprecationWarning: `nbconvert.exporters.exporter_locator` is deprecated in favor of `nbconvert.exporters.base` since nbconvert 5.0.
    DeprecationWarning)

reco_utils/recommender/rbm/rbm.py:452
  /data/home/recocat/notebooks/miguel/Recommenders/reco_utils/recommender/rbm/rbm.py:452: DeprecationWarning: invalid escape sequence \s
    """

tests/unit/test_notebook_utils.py::test_is_jupyter
  /anaconda/envs/reco_base/lib/python3.6/site-packages/jupyter_client/session.py:371: DeprecationWarning: Session._key_changed is deprecated in traitlets 4.1: use @observe and @unobserve instead.
    def _key_changed(self):

-- Docs: https://docs.pytest.org/en/latest/warnings.html
====================================================== slowest test durations ======================================================
1294.08s call     tests/unit/test_notebooks_python.py::test_wikidata_runs
217.83s call     tests/unit/test_notebooks_python.py::test_vw_deep_dive_runs
120.90s call     tests/unit/test_notebooks_python.py::test_rlrmc_quickstart_runs
45.77s call     tests/unit/test_notebooks_python.py::test_lightgbm
40.65s call     tests/unit/test_notebooks_python.py::test_surprise_deep_dive_runs
19.60s call     tests/unit/test_notebooks_python.py::test_sar_deep_dive_runs
16.61s call     tests/unit/test_notebooks_python.py::test_baseline_deep_dive_runs
12.61s call     tests/unit/test_notebooks_python.py::test_sar_single_node_runs
3.51s call     tests/unit/test_notebooks_python.py::test_template_runs
2.55s call     tests/unit/test_notebook_utils.py::test_is_jupyter

(0.00 durations hidden.  Use -vv to show these durations.)
===================================== 10 passed, 153 deselected, 6 warnings in 1777.51 seconds 
```

almudenasanz commented:
I have changed the default parameters of the notebook to do the sampling, and the test have passed. Seems like the injection of the parameters for the test was not working.
